portal/server/controllers/login/check-your-email.js

```diff
@@ -0,0 +1,48 @@
+const api = require('../../api');
+const { obscureEmail } = require('../../utils/obscure-email');
+
+module.exports.renderCheckYourEmailPage = (req, res) => {
+  const { session: { numberOfSendSignInLinkAttemptsRemaining, userEmail } } = req;
+
+  if (typeof numberOfSendSignInLinkAttemptsRemaining !== 'number') {
+    console.error('Number of send sign in link attempts remaining was not present in the session.');
+    return res.render('_partials/problem-with-service.njk');
+  }
+
+  if (numberOfSendSignInLinkAttemptsRemaining <= 0) {
+    return res.render(
+      'login/we-have-sent-you-another-link.njk',
+      { signInLinkTargetEmailAddress: obscureEmail(userEmail) },
+    );
+  }
+
+  if (numberOfSendSignInLinkAttemptsRemaining === 1) {
+    return res.render('login/new-sign-in-link-sent.njk');
+  }
+
+  return res.render('login/check-your-email.njk');
+};
+
+module.exports.sendNewSignInLink = async (req, res) => {
+  const { session: { numberOfSendSignInLinkAttemptsRemaining } } = req;
+
+  if (!numberOfSendSignInLinkAttemptsRemaining || numberOfSendSignInLinkAttemptsRemaining < 0) {
+    console.error(`User attempted to send a new sign in link without any attempts remaining. Attempts remaining count is ${numberOfSendSignInLinkAttemptsRemaining}.`);
+    return res.status(403).render('_partials/problem-with-service.njk');
+  }
+
+  // TODO DTFS2-6770: Should we worry about concurrency?
+  // TODO DTFS2-6770: Should we move this count to the database?
```

Comment:
I think this will have to be moved to the database (maybe if we go ahead with the changes as a result of webmail). What do you think?